src/app/workflow/dag/dag.component.html

```diff
@@ -21,9 +21,12 @@
         <mat-icon class="alert-warning-icon">warning</mat-icon>&nbsp;DAG display is still experimental for Nextflow.
       </mat-card>
     </div>
-    <div fxFlex *ngIf="dagType === 'classic' && (isNFL$ | async) === false && (dagResult$ | async) === false">
+    <div
+      fxFlex
+      *ngIf="dagType === 'classic' && (isNFL$ | async) === false && ((dagResult$ | async) === null || (dagResult$ | async | json) === '{}')"
```

Comment:
Shouldn't `((dagResult$ | async) === null || (dagResult$ | async | json) === '{}')` instead be `(missingTool$ | async)`?

Otherwise, there is the hypothetical case where the API response is something like `{nodes: [], edges: []}`, which wouldn't meet this `*ngIf` condition, and the code on line 40 would never get evaluated and that HTML never displayed.

Reply:
Good point! Then I suppose the type of warning displayed (missing the descriptor or missing required tools) would mainly depend on the value of `selectedVersion?.valid`.